packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -42,20 +71,24 @@ class _DepthFirstChildIterator implements Iterator<Element> {
       return false;
 
     _current = _stack.removeLast();
-    // Stack children in reverse order to traverse first branch first
-    _stack.addAll(_reverseChildrenOf(_current, skipOffstage));
+    _fillChildren(_current);
 
     return true;
   }
 
-  static Iterable<Element> _reverseChildrenOf(Element element, bool skipOffstage) {
+  void _fillChildren(Element element) {
     assert(element != null);
-    final List<Element> children = <Element>[];
+    // If we did not have to follow LTR order and could instead use RTL,
```

Comment:
Can you add a unit test that asserts LTR order to ensure we preserve that in the future as well?